test/lightning_web/live/workflow_live_test.exs

```diff
@@ -153,6 +154,160 @@ defmodule LightningWeb.WorkflowLiveTest do
+               },
+               "50 * * * *"
+             ) == "34 * * * *"
+
+      assert LightningWeb.JobLive.CronSetupComponent.get_cron_expression(
+               %{
+                 frequency: :daily,
+                 hour: "00",
+                 minute: "34",
+                 monthday: "01",
+                 weekday: "01"
+               },
+               "50 * * * *"
+             ) == "34 00 * * *"
+
+      assert LightningWeb.JobLive.CronSetupComponent.get_cron_expression(
+               %{
+                 frequency: :weekly,
+                 hour: "00",
+                 minute: "34",
+                 monthday: "01",
+                 weekday: "01"
+               },
+               "50 * * * *"
+             ) == "34 00 * * 01"
+
+      assert LightningWeb.JobLive.CronSetupComponent.get_cron_expression(
+               %{
+                 frequency: :monthly,
+                 hour: "00",
+                 minute: "34",
+                 monthday: "01",
+                 weekday: "01"
+               },
+               "50 * * * *"
+             ) == "34 00 01 * *"
+
+      assert LightningWeb.JobLive.CronSetupComponent.get_cron_expression(
+               %{
+                 frequency: :custom,
+                 hour: "00",
+                 minute: "34",
+                 monthday: "01",
+                 weekday: "01"
+               },
+               "50 * * * *"
+             ) == "50 * * * *"
+    end
+
+    test "cron_setup_component wired to job-form and works", %{
+      conn: conn,
+      project: project,
+      job: job
+    } do
+      {:ok, view, html} =
+        live(
+          conn,
+          Routes.project_workflow_path(conn, :edit_job, project.id, job.id)
+        )
+
+      assert html =~ project.name
+
+      assert has_element?(view, "#job-form")
+
+      assert view
+             |> form("#job-form",
+               job_form: %{trigger_type: "cron"}
+             )
+             |> render_change()
+
+      assert view
+             |> element("#frequency")
+             |> render()
+             |> parse()
+             |> xpath(~x"option/text()"l) == [
+               'Every hour',
+               'Every day',
+               'Every week',
+               'Every month',
+               'Custom'
+             ]
+
+      assert view
+             |> element("#frequency")
+             |> render_change(%{cron_component: %{frequency: "monthly"}})
+
+      assert view
+             |> element("#minute")
+             |> render_change(%{cron_component: %{minute: "05"}})
+
+      view |> form("#job-form") |> render_submit()
+
+      assert_patch(view, Routes.project_workflow_path(conn, :show, project.id))
+
+      view
+      |> render_patch(
+        Routes.project_workflow_path(conn, :edit_job, project.id, job.id)
+      )
+
+      assert view
+             |> has_element?("#frequency option[selected][value=monthly]"),
+             "Should have the option that was previously selected"
+    end
```

Comment:
@elias-ba I added a test here to show where the feature doesn't work reliably, whenever I change the value, save the job and return the value is always 'hourly'.

Reply:
@stuartc I think this is now fixed with the code I pushed lately. Sorry for this interference